patches/server/0967-General-ItemMeta-fixes.patch

```diff
@@ -1375,7 +1384,39 @@ index 7f9182809f6e67ff571db0f365bc7e05f600775a..01c49df291f721bab3acb788ff2f2787
  
          List<MobEffectInstance> effectList = new ArrayList<>();
          if (this.customEffects != null) {
-@@ -280,12 +280,12 @@ class CraftMetaPotion extends CraftMetaItem implements PotionMeta {
+@@ -189,25 +189,24 @@ class CraftMetaPotion extends CraftMetaItem implements PotionMeta {
+     public boolean addCustomEffect(PotionEffect effect, boolean overwrite) {
+         Preconditions.checkArgument(effect != null, "Potion effect cannot be null");
+ 
+-        int index = this.indexOfEffect(effect.getType());
+-        if (index != -1) {
+-            if (overwrite) {
++        // Paper start - support multiple effects of the same type
++        if (overwrite) {
++            int index = this.indexOfEffect(effect.getType());
++            if (index != -1) {
++        // Paper end - support multiple effects of the same type
+                 PotionEffect old = this.customEffects.get(index);
+                 if (old.getAmplifier() == effect.getAmplifier() && old.getDuration() == effect.getDuration() && old.isAmbient() == effect.isAmbient()) {
```

Comment:
Actually now that multiple effects of the same type is possible it might be better to do like the other methods and remove all the previous effects with the same type when overwrite is true and set the new effect at the first index found. I think the current behavior (pr) is confusing since overwrite will let other effects of the same type and will only override the first effect found.

Reply:
Implemented what you suggested.